File: java_solutions/MergeTwoSortedLists.java
package leetCode;

public class MergeTwoSortedLists {
	public static void main (String [] args){
		
	}
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode head = null;
        ListNode current = null;
        while (l1 != null && l2 != null){
        	if (l1.val < l2.val){
        		if (head == null){
        			head = l1;
        			current = head;
        			l1 = l1.next;
        		} else {
        			current.next = l1;
        			current = current.next;
        			l1 = l1.next;
        		}
        	} else {
        		if (head == null){
        			head = l2;
        			current = head;
        			l2 = l2.next;
        		} else {
        			current.next = l2;
        			current = current.next;
        			l2 = l2.next;
        		}
        	}
        }
        if (head == null){
        	if (l1 == null && l2 == null){
        		return null;
        	} else if (l1 == null){
        		head = l2;
        	} else {
        		head = l1;
        	}
        } else {
        	if (l1 == null){
        		current.next = l2;
        	} else {
        		current.next = l1;
        	}
        }
        return head;
    }
}

File: python_solutions/merge_two_sorted_lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """

        merged = None

        if l1 is None or l2 is None:
        	if l1 is None:
        		return l2
        	if l2 is None:
        		return l1

        if l1.val < l2.val:
        	merged = l1
        	l1 = l1.next
        else:
        	merged = l2
        	l2 = l2.next

        merged_tail = merged

        while l1 is not None and l2 is not None:
        	if l1.val < l2.val:
        		merged_tail.next = l1
        		l1 = l1.next
        	else:
        		merged_tail.next = l2
        		l2 = l2.next
        	merged_tail = merged_tail.next

        if l1 is None:
        	merged_tail.next = l2
        else:
        	merged_tail.next = l1
        return merged
